db/tags/extension.hh

```diff
@@ -37,4 +37,9 @@ private:
     std::map<sstring, sstring> _tags;
 };
 
+// Information whether the view updates are synchronous is stored using the
+// SYNCHRONOUS_VIEW_UPDATES_TAG_KEY tag. Value of this tag is a stored as a
+// serialized boolean value ("true" or "false")
+static const sstring SYNCHRONOUS_VIEW_UPDATES_TAG_KEY("system:synchronous_view_updates");
```

Comment:
Currently in in Alternator, you can't tag a view table, just the base table...
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Tagging.html says "Any local secondary indexes (LSI) and global secondary indexes (GSI) related to tagged tables are labeled with the same tags automatically." but we haven't implemented that. We could, I guess.

Thinking about this made me ask myself a question: do we really want "synchronous updates" to be based on a tag on the view table? Wouldn't it be even more convenient to put a tag on the base table, and it will apply to writes to all its views? Or do we envision a situation where a user might want to make some views synchronous and some not?

Reply:
When I thought about https://github.com/scylladb/scylla/issues/10545 I thought that this granularity may be useful. I imagined a case when somebody wants to create X materialized views that are just "best effort" and can be rebuilt or even dropped if necessary, but they also care about consistency for a single specific view. Then, only this view is marked as synchronous and we only "pay" with availability for this one, while the rest is still asynchronous and does not cause the base write to fail. Since synchronous updates sent over the network are costly, I figured it's better to have higher granularity, especially that the implementation is not particularly complicated.

Reply:
I see. It makes sense. We could also allow *both* options (if you tag the base table, it applies to all views, or you can tag only an individual view) but that can get complicated.
As long as we can only tag the view, we should be aware that doing this is not currently possible in Alternator. I'll open an issue.

Reply:
I opened https://github.com/scylladb/scylla/issues/10786 about the tagging of materialized views in Alternator (which should be allowed in one form or another). But this doesn't block this patch - we can start supporting this feature before Alternator does.